### PR TITLE
Add a type-safe wrapper for open flags

### PR DESCRIPTION
**Pull request by passaro**
## Description of change

Introduce `OpenFlags`, a type-safe wrapper for the flags provided to the `open` FUSE request. The new type is for internal use in `S3FileSystem` and replaces `i32` and `libc` constants.

Note that the set of allowed flags is not exhaustive (some flags are platform-dependent) and intended to be extended as required.

The `flags` module also provides the `libc_flags` helper macro, which should be handy when we wrap other flags types.

## Does this change impact existing behavior?

No functional changed.

## Does this change need a changelog entry in any of the crates?

No.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

**Review by dannycjones — COMMENTED** [review]
This is great, thank you for adding this!

Just one thing on the testing because I'm a bit paranoid.

**Comment by passaro** [comment]
I just realized `libc::O_RDONLY` is obviously not handled well. Let me think about this a bit more. Back to draft.

**Comment by passaro** [comment]
Removed values for `O_RDONLY` and `O_ACCMODE` and added comments.

File: mountpoint-s3/Cargo.toml [hunk 1]
```diff
@@ -16,6 +16,7 @@ async-channel = "2.1.1"
 async-lock = "3.3.0"
 async-trait = "0.1.57"
 bincode = "1.3.3"
+bitflags = "2.5.0"
```
[the remark this answers is not in the document]
Comment (passaro) [reply mountpoint-s3/Cargo.toml:19]: `2.5.0` is already included through other deps. I'm planning to update in a separate change.

**Review by dannycjones — APPROVED** [review]
LGTM!

There's nothing stopping us from having something like this if you prefer, but I'm happy as is.

```rust
impl OpenFlags {
    pub fn is_read_only_mode() -> bool { .. }
    pub fn is_write_only_mode() -> bool { .. }
    pub fn is_read_write_mode() -> bool { .. }
}
```

**Comment by passaro** [comment]
> LGTM!
> 
> There's nothing stopping us from having something like this if you prefer, but I'm happy as is.
> 
> ```rust
> impl OpenFlags {
>     pub fn is_read_only_mode() -> bool { .. }
>     pub fn is_write_only_mode() -> bool { .. }
>     pub fn is_read_write_mode() -> bool { .. }
> }
> ```

Yes. Or even an `AccessMode` enum, which would work very well with the (single) usage. I'll merge as-is, we can add to it later.

**Pull request merged**